### PR TITLE
[Go] Generate test files

### PR DESCRIPTION
**Pull request by gcatanese**
@antihax (2017/11) @grokify (2018/07) @kemokemo (2018/09) @jirikuncar (2021/01) @ph4r5h4d (2021/04)

Updated Go codegen to generate the test files as well, adding a new `api_test.mustache`. See #13559.

I have provided a basic implementation of a test, and added the skip option by default (I think it is very helpful to have a basic test implementation to start with, which can be extended and enabled by developer)

**How to validate**

Generate Go client and verify the test files have been created

```
java -jar /openapi-generator/modules/openapi-generator-cli/target/openapi-generator-cli.jar  generate \
-i openapi.yaml -g go  -p packageName=Management -o src/folder
```

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java [hunk 1]
```diff
@@ -100,10 +100,11 @@ public GoClientCodegen() {
 
         apiTemplateFiles.put("api.mustache", ".go");
         modelTemplateFiles.put("model.mustache", ".go");
+        apiTestTemplateFiles.put("api_test.mustache", ".go");
         modelDocTemplateFiles.put("model_doc.mustache", ".md");
         apiDocTemplateFiles.put("api_doc.mustache", ".md");
 
-        // default HIDE_GENERATION_TIMESTAMP to true
+                // default HIDE_GENERATION_TIMESTAMP to true
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java:107]: minor: fix the indention pls

Comment (gcatanese) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java]: Done

**Comment by janhartigan** [comment]
@gcatanese @wing328

We've noticed this change seems to create a broken import error. Not sure if this is a bug or if we're doing something wrong.

Screen Shot 2022-11-01 at 15 45 56

**Comment by gcatanese** [comment]
@janhartigan What is in the generated `go.mod`?

**Comment by janhartigan** [comment]
@gcatanese 

```
module github.com/goartica/product-service-northstar-go-client

go 1.13

require (
	github.com/stretchr/testify v1.4.0
	golang.org/x/oauth2 v0.0.0-20210323180902-22b0adad7558
)
```

**Comment by gcatanese** [comment]
@wing328 I have created a new PR #13905 to address this

@janhartigan This is now corrected, thanks for pointing it out. Meanwhile (if it helps) you can fix this manually in the generated test file(s) correcting the import with:
```
openapiclient "github.com/GIT_USER_ID/GIT_REPO_ID"
```

**Pull request merged**